Change Kubeflow Distributions Table (#3693)
kubeflow/website: content/en/docs/started/installing-kubeflow.md

+++

## What is Kubeflow?

Kubeflow is an end-to-end Machine Learning (ML) platform for Kubernetes, it provides components for each stage in the ML lifecycle, from exploration through to training and deployment.
Operators can choose what is best for their users, there is no requirement to deploy every component.

Learn more about Kubeflow in the [Introduction](/docs/started/introduction/) and
[Architecture](/docs/started/architecture/) pages.

## How to install Kubeflow?

Anywhere you are running Kubernetes, you should be able to run Kubeflow.
There are two primary ways to install Kubeflow:

1. [**Packaged Distributions**](#packaged-distributions-of-kubeflow)

> 1. [**Raw Manifests**](#raw-kubeflow-manifests) <sup>(advanced users)</sup>

Reviewer: ```suggestion
2. [**Raw Manifests**](#raw-kubeflow-manifests) <sup>(advanced users)</sup>
```

Reviewer 2: Markdown allows to use the same number twice, and it makes it easier to add new elements without updating multiple lines.

Reviewer 2: And I can't reply to the above comment, so here is a link to my in-thread reply. https://github.com/kubeflow/website/pull/3693/files#r1521923323

Reviewer: I see. SGTM

### How to install Kubeflow?

Anywhere you are running Kubernetes, you should be able to run Kubeflow.
There are two primary ways to install Kubeflow:

1. [**Packaged Distributions**](#packaged-distributions-of-kubeflow)
1. [**Raw Manifests**](#raw-kubeflow-manifests) <sup>(advanced users)</sup>

<a id="packaged-distributions"></a>
<a id="install-a-packaged-kubeflow-distribution"></a>

## Packaged Distributions of Kubeflow

Packaged distributions are maintained by various organizations and typically aim to provide
a simplified installation and management experience for Kubeflow.

{{% alert title="Note" color="warning" %}}
Packaged distributions are developed and supported by their respective maintainers.
The Kubeflow community <strong>does not endorse or certify</strong> any specific distribution.

In the near future, there are plans to introduce <a href="https://github.com/kubeflow/community/blob/master/proposals/kubeflow-conformance-program-proposal.md">conformance testing for distributions</a>, you may track progress on this initiative by following <a href="https://github.com/kubeflow/kubeflow/issues/6485">kubeflow/kubeflow#6485</a>.
{{% /alert %}}

> Some distributions can be deployed on all certified Kubernetes

Reviewer 2: I think we should move this above the "warning" so that it becomes part of the introduction sentence for this section.

Here is my proposal for the intro sentence:

```md
Packaged distributions are maintained by various organizations and typically aim 
to provide a simplified installation and management experience for Kubeflow.
Some distributions can be deployed on  [all certified Kubernetes distributions](https://kubernetes.io/partners/#conformance), 
while others target a specific platform (e.g. EKS or GKE).
```

Author: Sounds good.

distributions <sup>[<a href="https://kubernetes.io/partners/#conformance">1</a>]</sup>, some of them
can be deployed only in specific Kubernetes environment (e.g. EKS or GKE).

> The following table lists distributions which are <em>maintained</em> by their respective maintainers.

Reviewer 2: I think this is worded a bit strange, let's highlight that maintained means "active" like the current page:

```suggestion
The following table lists __active distributions__ which are _currently maintained_ by their respective maintainers:
```

Author: I removed `active` and `currently` on purpose. As I said before, some users might want to use distributions without active maintenance from respective owners (e.g. Kubeflow on AWS). So to avoid confusion, I simplified this sentence.

> <div class="table-responsive distributions-table">

Reviewer 2: Based on the discussions in today's community meeting, I think a good compromise is:

1. Keep a single table for all distributions.
1. Re-add the "Distribution Name" column:
      - But leave it empty for all distributions currently called `"Kubeflow on XXX"`.
      - Have each distribution make a PR to update the table, once they decide how they want to brand themselves (within the confines of the Kubeflow trademark rules)
1. Order the columns (from left to right):
     - Platform
     - Kubeflow Version
     - Distribution Name
     - Maintainer
     - Links
1. Sort the rows of the table alphabetically (starting from the left-most column).

__For example:__

Platform | Kubeflow Version |  Distribution Name | Maintainer | Links
--- | --- | --- | --- | ---
All Certified Kubernetes Distributions | 1.8.0<sup>[Release Notes]</sup> | Charmed Kubeflow | Canonical | Website
All Certified Kubernetes Distributions | 1.7.0<sup>[Version Matrix]</sup> | deployKF | Aranui Solutions | Website
Google Kubernetes Engine (GKE) | 1.8.0<sup>[Release Notes]</sup> | - | Google Cloud | Website
IBM Cloud Kubernetes Service (IKS) | 1.8.0<sup>[Release Notes]</sup> | - | IBM Cloud | Website
... | ... | ... | ...

Reviewer 3: I am supportive of this proposal.    One comment - The initial sort might be by Kubeflow Version (rather than by Platform/alpha).  This rewards the distributions that stay up to date.   If the Kubeffow Version is not the 1st column, this ordering might be confusing to the reader.   If the Version is the 1st column, that might be awkward looking.  We might add a comment in the Header - Kubeflow Distributions sorted by latest Kubeflow Version supported.

Reviewer 2: @jbottum from a user perspective, they will be focused on finding a distribution that works on their platform, not finding whatever the "latest" version for other platforms might be.

I think that's why @andreyvelich suggested sorting by the target platform.

Author: @thesuperzapper 
>Re-add the "Distribution Name" column:

What if distribution owners don't want to give name for their distribution ?

>Distribution Name order

From my perspective, maintainer column is more important for users than distribution name.

Reviewer 2: > What if distribution owners don't want to give name for their distribution ?

I think we can just put it as `-` in those cases, like when it's not really a distribution (i.e. a ML Platform), but a simple README of how to use the Kubeflow manifests on their platform.

> From my perspective, maintainer column is more important for users than distribution name.

I think most distribution maintainers would prefer to "brand" their distribution as its name, rather than their company name, so my preference is to list "Distribution Name" before "Maintainer",

However, I am open to hearing others' thoughts on this.

+++

> ## What is Kubeflow?

Reviewer 2: I think this looks better as H3, like the `How to install Kubeflow?` below it:

```suggestion
### What is Kubeflow?
```

Author: We might want to keep titles with the same font if it is not sub-title. Like in Architecture page: https://deploy-preview-3693--competent-brattain-de2d6d.netlify.app/docs/started/architecture/#conceptual-overview.
So all title fonts on the same level will be consistent.

So if you don't want to make `How to install Kubeflow? ` subtitle, I can change it as follows:
```
## How to install Kubeflow? 
```

Author: I will make them both H2. If we make the first title H3, website will show outline not accurate 
<img width="293" alt="Screenshot 2024-03-08 at 13 07 00" src="https://github.com/kubeflow/website/assets/31112157/9a4181c1-7b3c-4742-ae77-a26bd45b6252">

Kubeflow is an end-to-end Machine Learning (ML) platform for Kubernetes, it provides components for each stage in the ML lifecycle, from exploration through to training and deployment.
Operators can choose what is best for their users, there is no requirement to deploy every component.

> Learn more about Kubeflow in the [Introduction](/docs/started/introduction/) and

Reviewer 2: I think we can put a newline here for visual separation:

```suggestion

Learn more about Kubeflow in the [Introduction](/docs/started/introduction/) and
```

Reviewer: This seems not to be addressed.

Reviewer 2: This is in reply to a suggestion for a newline.

It was added a few commits ago, I can see it on the test website:
- https://deploy-preview-3693--competent-brattain-de2d6d.netlify.app/docs/started/installing-kubeflow/

Reviewer: SGTM

1. [**Packaged Distributions**](#packaged-distributions-of-kubeflow)
1. [**Raw Manifests**](#raw-kubeflow-manifests) <sup>(advanced users)</sup>

<a id="packaged-distributions"></a>
<a id="install-a-packaged-kubeflow-distribution"></a>

## Packaged Distributions of Kubeflow

Packaged distributions are maintained by various organizations and typically aim to provide
a simplified installation and management experience for Kubeflow.

{{% alert title="Note" color="warning" %}}
Packaged distributions are developed and supported by their respective maintainers.
The Kubeflow community <strong>does not endorse or certify</strong> any specific distribution.

In the near future, there are plans to introduce <a href="https://github.com/kubeflow/community/blob/master/proposals/kubeflow-conformance-program-proposal.md">conformance testing for distributions</a>, you may track progress on this initiative by following <a href="https://github.com/kubeflow/kubeflow/issues/6485">kubeflow/kubeflow#6485</a>.
{{% /alert %}}

Some distributions can be deployed on all certified Kubernetes
distributions <sup>[<a href="https://kubernetes.io/partners/#conformance">1</a>]</sup>, some of them

> can be deployed only in specific Kubernetes environment (e.g. EKS or GKE).

Reviewer 2: If we are going to make the distinction between distributions for all Kubernetes, and ones for only a specific platform, I think it makes sense to split the tables up into "generic" and "specific" with different columns:

__"Generic" Table:__

```md
Some Kubeflow distributions can be deployed on [all certified Kubernetes
distributions](https://kubernetes.io/partners/#conformance):
```

Distribution Name | Kubeflow Version | Maintainer | Links
--- | --- | --- | ---
Charmed Kubeflow | 1.8.0 | Canonical | Website
deployKF | 1.7.0 | Aranui Solutions | Website

---

__"Specific" Table:__

```md
Other Kubeflow distributions target a specific Kubernetes platform:
```

Platform | Kubeflow Version | Maintainer | Links
--- | --- | --- | ---
Google Kubernetes Engine (GKE) | 1.8.0 | Google Cloud | Website
IBM Cloud Kubernetes Service (IKS) | 1.8.0 | IBM Cloud | Website
... | ... | ... | ...

Author: Yes, we discussed this before, so your suggestion @thesuperzapper is to name Distributions as:
- Genetic Distributions
- Specific Distributions

What do others think about it @ca-scribner @kubeflow/kubeflow-steering-committee @juliusvonkohout ?

Reviewer 4: Probably yes. Which table should go first? Maybe the specific ones?

Author: I will raise this question on tomorrow's call.

Reviewer 2: @andreyvelich I think putting the "generic" table first is more in the spirit of Kubeflow as a "generic" platform.

I worry that listing the "specific" distributions first will imply that the first platform on that table is the "recommended platform for Kubeflow". (We are already trying to shake the perception that Kubeflow is a Google-focused product).

Also, users will already gravitate toward the distribution targeting their specific platform, so it just doubly disadvantages the generic distributions by putting them second.

However, I understand that I may be a little biased (I maintain deployKF), so I am happy to see what others think.

Reviewer 2: Based on today's community meeting, I think the consensus was to keep a single table, and to re-add the "Distribution Name" column (but keep it empty for distributions initially, until they make a PR to update the website with a non `"Kubeflow on XXXX"` name.

See my comment here for more details: https://github.com/kubeflow/website/pull/3693#discussion_r1513480637

The following table lists distributions which are <em>maintained</em> by their respective maintainers.

> <div class="table-responsive distributions-table-active">

Reviewer 2: You forgot to port the style changes which added this CSS class:

See the `./assets/scss/_styles_project.scss` file diff form: https://github.com/kubeflow/website/pull/3643/files

Author: Nice catch!

  <table class="table table-bordered">
    <thead>
      <tr>
        <th>Target Platform</th>
        <th>Kubeflow Version</th>
        <th>Maintainer / Distribution Name</th>
        <th>Link</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          All Certified Kubernetes Distributions
        </td>
        <td>

>           {{% canonical/latest-version %}}

Reviewer 2: We should add the link to Charmed Kubeflow's release notes, like the other distributions:

```suggestion
          {{% canonical/latest-version %}} <sup>[<a href="https://charmed-kubeflow.io/docs/release-notes">Release Notes</a>]</sup>
```

        </td>
      </tr>
      <tr>
        <td>
          Google Kubernetes Engine (GKE)
        </td>
        <td>
          {{% gke/latest-version %}} <sup>[<a href="https://googlecloudplatform.github.io/kubeflow-gke-docs/docs/changelog/">Release Notes</a>]</sup>
        </td>
        <td>
          Google Cloud
        </td>
        <td>
          <a href="https://googlecloudplatform.github.io/kubeflow-gke-docs">Website</a>
        </td>
      </tr>
      <tr>
        <td>
          IBM Cloud Kubernetes Service (IKS)
        </td>
        <td>
          {{% iks/latest-version %}} <sup>[<a href="https://github.com/IBM/manifests/releases/tag/v{{% iks/latest-version %}}">Release Notes</a>]</sup>
        </td>
        <td>
          IBM Cloud
        </td>
        <td>
          <a href="https://ibm.github.io/manifests/">Website</a>
        </td>
      </tr>
      <tr>
        <td>
          Nutanix Kubernetes Engine
        </td>
        <td>
          {{% nutanix/latest-version %}}
        </td>
        <td>
          Nutanix
        </td>
        <td>
          <a href="https://nutanix.github.io/kubeflow-manifests">Website</a>
        </td>
      </tr>
      <tr>
        <td>

>           Amazon Elastic Kubernetes Service (EKS)

Reviewer 2: I still think we should consider removing this distribution, AWS has indicated they are no longer supporting it, so it's a bit crazy to include it (which will make people adopt a distribution that will not get any updates):

Here is their announcement that they will not support 1.8:

- https://github.com/awslabs/kubeflow-manifests/issues/794
- And looking by [commits on the repo](https://github.com/awslabs/kubeflow-manifests) I am pretty sure they are not maintaining it.

Reviewer 2: @ananth102 you're the only person to commit on the `awslabs/kubeflow-manifests` repo in the last 5 months, what is the status of that repo?

Author: I have different thoughts here.
Since EKS is one of the most popular Kubernetes environment and many folks are considering to use Kubeflow distribution that was created and was maintained by AWS folks, we might want to keep it for a while.
Even that AWS was not participating in the recent Kubeflow release.
In the future, AWS might work on this again.

Also, as I said here: https://github.com/kubeflow/website/pull/3693#issuecomment-1976597192, we should list distributions with `release-7` support.

cc @kubeflow/kubeflow-steering-committee @surajkota @jsitu777 @ryansteakley

        </td>
        <td>
          Microsoft Azure
        </td>
        <td>
          1.7.0 <sup>[<a href="https://www.deploykf.org/releases/version-matrix/#kubeflow-tools">Version Matrix</a>]</sup>
          <a href="https://azure.github.io/kubeflow-aks/main">Website</a>
        </td>
      </tr>
      <tr>
        <td>Kubeflow on Oracle Container Engine for Kubernetes</td>
        <td>Oracle</td>
        <td>
          Oracle Container Engine for Kubernetes (OKE)
          VMware vSphere
        </td>
        <td>
          <a href="https://github.com/oracle-devrel/kubeflow-oke">Website</a>
          {{% vmware/latest-version %}}
        </td>
        <td>
          1.6.0
          VMware
        </td>
      </tr>
      <tr>
        <td>Kubeflow on vSphere</td>
        <td>VMware</td>
        <td>VMware vSphere</td>
        <td>
          <a href="https://vmware.github.io/vSphere-machine-learning-extension/">Website</a>
        </td>
        <td>
          1.6.1
        </td>
      </tr>
    </tbody>
  </table>
</div>

> ### Legacy Distributions

Reviewer 2: You have accidentally left all the legacy distributions (and placed them under the maintained table), if they aren't in a separate table, let's remove them.

We need to remove:

- Argoflow
- Kubeflow on OpenShift
- Arrikto Enterprise Kubeflow

Author: I will remove Argoflow + Arrikto, but for OpenShift can we can confirmation from the RedHat team @rimolive @terrytangyuan @rareddy @kubeflow/red-hat ?

Reviewer 5: Red Hat is preparing to help with kubeflow 1.9 distribution testing. Let's keep it
